test/spec/idx/recoverPassword.ts

```diff
@@ -175,6 +178,57 @@ describe('idx/recoverPassword', () => {
     });
   });
 
+  it('returns a pending status with error message if a bad username was submitted', async () => {
+    const {
+      authClient,
+      identifyResponse,
+      identifyRecoveryResponse,
+    } = testContext;
+    
+    const username = 'incorrect@wrong.com';
+
+    // messages appear in the "rawIdxState"
+    const rawIdxState = RawIdxErrorFactory.build({
+      messages: IdxErrorMessagesFactory.build({
+        value: [
+          IdxErrorNoAccountWithUsernameFactory.build({}, {
+            transient: {
+              username
+            }
+          })
+        ]
+      })
+    });
+    const errorResponse = Object.assign({}, identifyRecoveryResponse, { rawIdxState });
+
+    jest.spyOn(identifyRecoveryResponse, 'proceed').mockResolvedValueOnce(errorResponse);
+    jest.spyOn(mocked.idx, 'start')
+      .mockResolvedValueOnce(identifyResponse);
+
+    const res = await recoverPassword(authClient, { username });
+    expect(res).toEqual({
+      status: IdxStatus.PENDING,
+      tokens: null,
+      error: undefined, // TOOD: is this expected?
```

Comment:
The current behaviour from the SDK is to expose all fields, and leave those without real values as `undefined`. I just found it's less code during implementation. We can change to just hide the field if it's not available.

Reply:
I'm just wondering if we should populate the "error" field when there is an error

Reply:
this field is for `AuthSDKError`, otherwise I was using `messages` field for idx backend messages to delivery full message information to the devs.